README.rst

```diff
@@ -0,0 +1,139 @@
+
+.. _Git LFS: https://git-lfs.github.com
+.. _DM Developer Guide: https://developer.lsst.io/git/git-lfs.html
+.. _Docker: https://www.docker.com
+.. _Docker Compose: https://docs.docker.com/compose/overview/
+
+Technology Summary
+==================
+
+The LSST Alert Distribution Service distributes alert packets, formatted using `Apache Avro`_ , using the `Apache Kafka`_ streaming platform.
+Each alert is packaged as a separate Kafka message.
+Schemata are not sent with the alerts: the consumer is assumed to receive a copy of the schema through some other mechanism (currently by cloning https://github.com/lsst-dm/sample-avro-alert).
+Alerts are packaged using the `Confluent Wire Format`_.
+This means that the first byte of the message received may be ignored, the next four constitute a “schema identifier” (which may be used to identify the schema used to write the packet) and the remainder constitute the alert data serialized in Avro format.
+Intepreting the packet in (for example) Python may then be done as follows::
+
+   import struct
+
+   raw_alert_bytes = ... # the data received from Kafka
+   intial_byte = struct.unpack("!b", raw_alert_bytes[0])  # should always be 0
+   schema_id = struct.unpack("!I", raw_alert_bytes[1:5])  # schema identifier
+   avro_bytes = raw_alert_bytes[5:]                       # feed to your Avro deserializer
+
+.. _Apache Avro: https://avro.apache.org
+.. _Apache Kafka: https://kafka.apache.org
+.. _Confluent Wire Format: https://docs.confluent.io/current/schema-registry/docs/serializer-formatter.html#wire-format
+
+Usage Instructions
+==================
+
+To deploy this service on a single host, first clone this repository, change into the cloned directory, and (if applicable) check out the relevant branch, then follow the instructions below.
+
+The service may also be deployed over multiple hosts using Docker Swarm.
+See ``swarm/README.md`` for instructions.
+
+Bring up Kafka and Zookeeper
+----------------------------
+
+From the alert_stream directory::
+
+   $ docker-compose up -d
+
+This will create a network named ``alert_stream_default`` with the default driver over which the other containers will connect and will start Kafka and Zookeeper.
+
+Build the Docker image
+----------------------
+
+From the ``alert_stream`` directory::
+
+   $ docker build -t "alert_stream" .
+
+and refer to it in subsequent run commands below for starting containers.
+
+This should now work::
+
+   $ docker run -it --rm alert_stream python bin/sendAlertStream.py -h
+
+You must rebuild your image every time you modify any of the code, unless you mount local code as a volume in the container.
+
+Start producing an alert stream
+-------------------------------
+
+Sample data is included in the ``data`` directory.
+You can also mount a local volume of data following the instructions below.
```

Comment:
Maybe we should say "You can also mount a local volume of data by replacing `$PWD/data` in the command below with an appropriate path to other alert data.